src/main/java/controllers/UserProfileController.java

```diff
@@ -226,4 +226,53 @@ private void runCustomizeBio() {
         String newBio = this.inOut.getInput("Enter what you'd like your new bio to be: ");
         this.runChangeBioDisplay(this.loginManager.getCurrUser(), newBio);
     }
+
+    /**
+     * Generate a selection of displayed posts' titles for the user to select.
+     *
+     * @param userPosts A hashmap of Posts representing a user's posts.
+     * @return a String representing all the posts of a user.
+     */
+    public String generateDisplayedPosts(HashMap<Integer, Post> userPosts) {
+        StringBuilder postsString = new StringBuilder("Enter a post number to view the post: ");
+
+        for (int i = 0; i < userPosts.size(); i++) {
+            postsString.append("\n");
+            postsString.append(i).append(" ");
+            postsString.append(userPosts.get(i).getRecipe().getTitle());
+        }
+        return postsString.toString();
+    }
+
+    /**
+     * Allow the user to act on a post selected.
+     *
+     * @param userPostsString     the string representing a user's posts
+     */
+    public void selectOnePost(String userPostsString, HashMap<Integer, Post> userPosts) {
+        PostController postController = new PostController(this.inOut, this.databaseManager, this.loginManager);
+        int postNumber = this.getSelectedPostInput(userPostsString);
+        Post selectedPost = userPosts.get(postNumber);
+        postController.browsePost(selectedPost);
+
+    }
+
+    /**
+     * Get the user to select a post to act on.
+     *
+     * @param userPostsString     the String representing the selection of displayed Posts' titles generated.
+     * @return an int representing the Post that the user selects.
+     */
+    private int getSelectedPostInput(String userPostsString) {
+        int postNumber = -700;
```

Comment:
Why -700?

Reply:
it was just a negative number to initialize postNumber